client/src/main/scala/io/finch/client/resources.scala

```diff
@@ -0,0 +1,27 @@
+package io.finch
+package client
+
+import com.twitter.finagle.httpx.{HeaderMap, Response}
+import com.twitter.io.Buf
+
+sealed trait Resource[A] { 
+  val headers: HeaderMap
+  val content: A
+  def as[B](implicit ev: A => Buf, d: DecodeResource[B]): Resource[B] = 
```

Comment:
Maybe I don't understand the intended usage yet, but implicit functions make me nervous. Would it be possible to provide this as a `def toBuf(a: A): Buf` on the trait?